### PR TITLE
Always optimize in/2 for literals

### PR DESCRIPTION
**Pull request by ericmj**

### CODE REVIEW

File: lib/mix/lib/mix/tasks/profile.fprof.ex [hunk 1]
```diff
@@ -126,7 +126,9 @@ defmodule Mix.Tasks.Profile.Fprof do
 
     # Start app after rewriting System.argv,
     # but before requiring and evaling
-    Mix.Task.run "app.start", args
+    unless opts[:start] == false do
```
Comment (michalmuskala) [inline lib/mix/lib/mix/tasks/profile.fprof.ex]: Maybe `if Keyword.get(opts, :start, true)` will be clearer?

Comment (josevalim) [inline lib/mix/lib/mix/tasks/profile.fprof.ex]: `unless "--no-start" in args` is the pattern we use throughout Mix for this one.

File: lib/elixir/test/elixir/kernel_test.exs [hunk 2]
```diff
@@ -194,6 +194,30 @@ defmodule KernelTest do
...
+             """
+             case(foo) do
+               _(foo in [1, 2]) ->
+                 :ok
+             end
+             """)
+  end
+
+  defp expand_to_string(ast) do
+    Macro.expand(ast, __ENV__)
```
Comment (josevalim) [inline lib/elixir/test/elixir/kernel_test.exs]: You should pipe all the way, starting with `ast`. :P

**Comment by ericmj** [comment]
Updated based on comments. Also added a commit that reduces the number of allocations in URI.percent/2.

**Pull request merged**